packages/ra-core/src/controller/useSortState.ts

```diff
@@ -64,7 +67,8 @@ export default (
     }, [initialSort.field, initialSort.order]); // eslint-disable-line react-hooks/exhaustive-deps
 
     return {
-        setSort: (field: string) => dispatch(field),
+        setSort: (sortValue: Sort) => dispatch(sortValue),
+        setSortField: (field: string) => dispatch(field),
```

Comment:
if you add a setSortField, you should probably also add a setSortOrder.

Also, this addition shows that you can call `dispatch` with both a stirng and an object, I think it's wrong.